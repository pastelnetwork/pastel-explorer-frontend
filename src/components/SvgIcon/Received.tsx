const Received = () => {
  return (
    <svg viewBox="0 0 69 65" fill="none" xmlns="http://www.w3.org/2000/svg">
      <path
        d="M31.8959 45.75C31.8959 50.3913 33.7396 54.8425 37.0215 58.1244C40.3034 61.4063 44.7546 63.25 49.3959 63.25C54.0372 63.25 58.4884 61.4063 61.7702 58.1244C65.0521 54.8425 66.8959 50.3913 66.8959 45.75C66.8959 41.1087 65.0521 36.6575 61.7702 33.3756C58.4884 30.0937 54.0372 28.25 49.3959 28.25C44.7546 28.25 40.3034 30.0937 37.0215 33.3756C33.7396 36.6575 31.8959 41.1087 31.8959 45.75Z"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2 10.75C2 13.0706 3.84374 15.2962 7.12563 16.9372C10.4075 18.5781 14.8587 19.5 19.5 19.5C24.1413 19.5 28.5925 18.5781 31.8744 16.9372C35.1563 15.2962 37 13.0706 37 10.75C37 8.42936 35.1563 6.20376 31.8744 4.56282C28.5925 2.92187 24.1413 2 19.5 2C14.8587 2 10.4075 2.92187 7.12563 4.56282C3.84374 6.20376 2 8.42936 2 10.75V10.75Z"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M37 19.5C37 24.3329 29.1658 28.25 19.5 28.25C9.83417 28.25 2 24.3329 2 19.5"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M37 23.875V10.75"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2 10.75V54.5C2 59.3329 9.83417 63.25 19.5 63.25C21.7038 63.2535 23.9023 63.0356 26.0625 62.5996"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2 45.75C2 50.5829 9.83417 54.5 19.5 54.5C21.7038 54.5035 23.9023 54.2856 26.0625 53.8496"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2 37C2 41.8329 9.83417 45.75 19.5 45.75C21.7038 45.7535 23.9023 45.5356 26.0625 45.0996"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M2 28.25C2 33.0829 9.83417 37 19.5 37C21.7038 37.0035 23.9023 36.7856 26.0625 36.3496"
        stroke="#219653"
        strokeWidth="3"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M47.5506 53.0839C47.7148 53.0816 47.8714 53.0147 47.9867 52.8977C48.102 52.7808 48.1666 52.6231 48.1666 52.4589L48.1666 49.2151C49.274 49.0916 51.0946 48.7742 52.9274 47.5875C55.0083 46.2401 56.8582 43.763 56.9012 39.6733C56.9018 39.6702 56.9038 39.6683 56.9044 39.6651L56.9028 39.6651C56.9032 39.6202 56.9166 39.5876 56.9166 39.5423C56.9165 39.3873 56.8587 39.2379 56.7546 39.1231C56.6506 39.0083 56.5075 38.9362 56.3533 38.9209C56.1991 38.9056 56.0447 38.9481 55.9201 39.0402C55.7954 39.1323 55.7095 39.2675 55.6788 39.4194C55.1841 41.8928 53.642 42.9889 51.9329 43.551C50.4904 44.0255 49.0781 44.0256 48.1666 43.9775L48.1666 40.7923C48.1666 40.6687 48.1299 40.5479 48.0612 40.4451C47.9925 40.3424 47.8949 40.2623 47.7808 40.215C47.6666 40.1677 47.5409 40.1553 47.4197 40.1794C47.2985 40.2035 47.1872 40.263 47.0997 40.3504L41.2664 46.1837C41.1492 46.3009 41.0834 46.4599 41.0834 46.6256C41.0834 46.7913 41.1492 46.9503 41.2664 47.0675L47.0997 52.9008C47.1588 52.9599 47.2292 53.0066 47.3066 53.038C47.384 53.0695 47.467 53.0851 47.5506 53.0839V53.0839ZM46.9166 50.9501L42.5921 46.6256L46.9166 42.301L46.9166 44.5903C46.9166 44.7459 46.9746 44.896 47.0793 45.0112C47.1841 45.1263 47.3281 45.1982 47.483 45.2128C48.3513 45.2941 50.3376 45.3916 52.3235 44.7384C53.2513 44.4332 54.1727 43.9196 54.9635 43.1856C54.3306 44.7332 53.3559 45.8202 52.2478 46.5377C50.4254 47.7177 48.3311 48.0046 47.4961 48.066C47.3386 48.0775 47.1913 48.1482 47.0839 48.2639C46.9764 48.3795 46.9166 48.5315 46.9166 48.6894L46.9166 50.9501Z"
        fill="#219653"
      />
      <rect x="59" y="56" width="20" height="20" transform="rotate(-180 59 56)" />
    </svg>
  );
};

export default Received;
