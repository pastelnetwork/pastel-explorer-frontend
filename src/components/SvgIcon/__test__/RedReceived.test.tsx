import { shallow } from 'enzyme';

import Received, { RedReceived } from '../Received';

describe('components/SvgIcon/Received', () => {
  const wrapper = shallow(<Received />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <svg>', () => {
    expect(wrapper.find('svg').length).toBe(1);
  });

  test('should render <rect>', () => {
    expect(wrapper.find('rect').length).toBe(1);
  });

  test('should render <path>', () => {
    expect(wrapper.find('path').length).toBe(9);
  });
});

describe('components/SvgIcon/Received/RedReceived', () => {
  const wrapper = shallow(<RedReceived />);

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should render <svg>', () => {
    expect(wrapper.find('svg').length).toBe(1);
  });

  test('should render <rect>', () => {
    expect(wrapper.find('rect').length).toBe(1);
  });

  test('should render <path>', () => {
    expect(wrapper.find('path').length).toBe(9);
  });
});
