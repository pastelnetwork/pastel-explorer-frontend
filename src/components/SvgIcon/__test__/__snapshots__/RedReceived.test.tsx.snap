// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SvgIcon/Received renders correctly 1`] = `ShallowWrapper {}`;

exports[`components/SvgIcon/Received/RedReceived renders correctly 1`] = `ShallowWrapper {}`;
